torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -495,3 +495,23 @@ def shard_parameter(
 
     # Now we can set the attribute appropriately.
     setattr(module, param_name, st)
+
+def gather(
+    sharded_tensor: ShardedTensor,
+    dst_rank: int = 0,
+    dst_tensor: Optional[torch.Tensor] = None,
+):
+    """
+    Creates an full :class:`Tensor` from sharded tensors by gather all shards.
+
+    Args:
+        sharded_tensor (:class `torch.distributed._sharded_tensor.ShardedTensor`): The sharded
+        tensor to gather.
+        dst_rank (int): The rank where full tensor is constructed
+        dst_tensor (:class `torch.Tensor` optional): The output full tensor. Need to and only
+        need to be provided on dst_rank
+
+    Returns:
+        None
+    """
+    return ShardedTensor._gather(sharded_tensor, dst_rank, dst_tensor)
```

Comment:
to confirm, we can remove this wrapper from __init__? curious what is the rule of thumb about when to put a wrapper in init and when not? thanks!

Reply:
Most of the methods in the init file are around creating a ShardedTensor and not necessarily methods that modify an existing ShardedTensor. If something acts on a instance of ShardedTensor its probably just have it as part of the ShardedTensor class instead of having another wrapper.